lib/cpp/src/thrift/transport/TServerSocket.cpp

```diff
@@ -659,6 +659,10 @@ shared_ptr<TSocket> TServerSocket::createSocket(THRIFT_SOCKET clientSocket) {
   }
 }
 
+boost::shared_ptr<THRIFT_SOCKET> TServerSocket::getInterruptSocket() {
```

Comment:
@jeking3 i can change the baseclass so the socket and the flag will be protected.